akka-actor-typed-tests/src/test/scala/docs/akka/typed/OOIntroSpec.scala

```diff
@@ -30,7 +30,7 @@ object OOIntroSpec {
     final case class GetSession(screenName: String, replyTo: ActorRef[SessionEvent]) extends RoomCommand
     //#chatroom-protocol
     //#chatroom-behavior
-    private final case class PublishSessionMessage(screenName: String, message: String) extends RoomCommand
+    final case class PublishSessionMessage(screenName: String, message: String) extends RoomCommand
```

Comment:
That is a rather important shortcoming. `private` like that is often used for differentiating between external (public) and internal messages.

It's documented here https://doc.akka.io/docs/akka/current/typed/actors.html#functional-style

> The behavior that we declare here can handle both subtypes of RoomCommand. GetSession has been explained already and the PublishSessionMessage commands coming from the session Actors will trigger the dissemination of the contained chat room message to all connected clients. But we do not want to give the ability to send PublishSessionMessage commands to arbitrary clients, we reserve that right to the internal session actors we create—otherwise clients could pose as completely different screen names (imagine the GetSession protocol to include authentication information to further secure this). Therefore PublishSessionMessage has private visibility and can’t be created outside the ChatRoom object.

so this change would break the documentation

Reply:
The error message dotty gives seems sensible:

```
[error] -- Error: /Users/luc/scala/akka/akka-actor-typed-tests/src/test/scala/docs/akka/typed/OOIntroSpec.scala:73:10
[error] 73 |      def apply(
[error]    |          ^
[error]    |non-private method apply in object SessionBehavior refers to private class PublishSessionMessage
```

```scala
object OOIntroSpec {
  object ChatRoom {
    private final case class PublishSessionMessage(screenName: String, message: String) extends RoomCommand
    object SessionBehavior {
      def apply(
          room: ActorRef[PublishSessionMessage],
          screenName: String,
          client: ActorRef[SessionEvent]): Behavior[SessionCommand] = ...
```

An alternative to my change is to change `object SessionBehavior` and make it `private`, that's probably better?

Reply:
ah, is it only because of the `object SessionBehavior`. Indeed better to make that `private`. Should have been that in the first place.